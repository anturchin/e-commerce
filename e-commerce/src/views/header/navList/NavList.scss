.nav-list {
    display: flex;
    flex-direction: row;
    align-items: center;
    list-style: none;
    padding-left: 0;
    gap: 35px;
}

@media (max-width: 550px) {
    .nav-list {
        flex-direction: column;
        padding: 1rem 0;
    }
}
