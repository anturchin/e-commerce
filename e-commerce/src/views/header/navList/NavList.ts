import { View } from '../../View';
import { NavItem } from '../navItem/NavItem';
import './NavList.scss';

const options: string[] = ['Home', 'Sign in', 'Sign up'];

export class NavList extends View {
    private navItems: NavItem[] = [];

    constructor() {
        super({ tag: 'ul', classNames: ['nav-list'] });
        this.setupNavList();
    }

    private setupNavList(): void {
        options.forEach((item) => {
            const navItem = new NavItem(item);
            this.navItems.push(navItem);
            this.viewHtmlElement.addInnerElement(navItem.getElement());
        });
    }
}
