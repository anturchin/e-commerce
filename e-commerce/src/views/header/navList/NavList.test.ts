import { describe, expect, test } from '@jest/globals';
import { NavList } from './NavList';

describe('NavList', () => {
    test('should match', () => {
        const list = new NavList();
        const renderedNavList = list.getElement();
        expect(renderedNavList.outerHTML).toMatchSnapshot();
    });
});
