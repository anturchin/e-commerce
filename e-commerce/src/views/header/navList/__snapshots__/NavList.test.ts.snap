// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList should match 1`] = `"<ul class="nav-list hidden"><li class="nav-list_item" data-nav-item="Home">Home</li><li class="nav-list_item" data-nav-item="Category">Category</li><li class="nav-list_item" data-nav-item="Sign in">Sign in</li><li class="nav-list_item" data-nav-item="Sign up">Sign up</li><li class="nav-list_item" data-nav-item="Profile">Profile</li><li class="nav-list_item" data-nav-item="About">About</li><li class="nav-list_item" data-nav-item="Logout">Logout</li></ul>"`;
